include/boost/geometry/algorithms/detail/intersection/areal_areal.hpp

```diff
@@ -0,0 +1,348 @@
+// Boost.Geometry
+
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_INTERSECTION_AREAL_AREAL_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_INTERSECTION_AREAL_AREAL_HPP
+
+
+#include <boost/core/ignore_unused.hpp>
+
+#include <boost/geometry/algorithms/detail/intersection/interface.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace intersection
+{
+
+
+template
+<
+    typename GeometryOut,
+    typename OutTag = typename detail::intersection::tag
+                        <
+                            typename geometry::detail::output_geometry_value
+                                <
+                                    GeometryOut
+                                >::type
+                        >::type
+>
+struct intersection_areal_areal_
```

Comment:
why the trailing `_` ?

Reply:
Because below the name `intersection_areal_areal` was already used. The original one. And I needed to distinguish between the original behavior and the handling of Tuples. Would you prefer a different name?